napari/layers/points/points_utils.py

```diff
@@ -0,0 +1,60 @@
+from typing import Dict, Tuple, Union
+
+import numpy as np
+from vispy.color.colormap import Colormap
+
+
+def dataframe_to_properties(dataframe) -> Dict[str, np.ndarray]:
+    """ Converts a dataframe to Points.properties formatted dictionary
+
+    Parameters
+    ----------
+    dataframe : DataFrame
+        The dataframe object to be converted to a properties dictionary
+
+    Returns
+    -------
+    dict[str, np.ndarray]
+        A properties dictionary where the key is the property name and the value
+        is an ndarray with the property value for each point.
+    """
+
+    properties = {col: np.asarray(dataframe[col]) for col in dataframe}
+    return properties
+
+
+def guess_continuous(property: np.ndarray) -> bool:
+    """Guess if the property is continuous (return True) or categorical (return False)"""
+    # if the property is a floating type, guess continuous
+    if issubclass(property.dtype.type, np.floating):
+        return True
```

Comment:
There's probably some threshold number of integer values, let's say either 15 or 127, beyond which integer values can probably also be considered continuous... But I'm happy to ignore this for now...

Reply:
That's a good point. I will set it at 15 for now. It's just a guess and the user is free to switch back and forth between cycles and colormaps, so I don't think we need to get the threshold exactly right 😛